go/vumitools/middleware.py

```diff
@@ -603,6 +604,151 @@ def handle_failure(self, failure, connector_name):
         return failure
 
 
+class ConversationMetricsMiddlewareConfig(MetricsMiddlewareConfig):
+
+    redis_manager = ConfigDict(
+        "Redis configuration parameters", default={}, static=True)
+
+    op_mode = ConfigText(
+        """
+        What mode to operate in, options are `passive` or `active`.
+        Defaults to passive.
+        *passive*:  assumes the middleware connector names are to be used as
+                    the names for metrics publishing.
+        *active*:   assumes that the individual messages are to be inspected
+                    for their `transport_name` values.
+
+        NOTE:   This does not apply for events or failures, the connectors
+                names are always used for those since those message types are
+                not guaranteed to have a `transport_name` value.
+        """,
+        default="passive", static=True)
+
+    metric_connectors = ConfigList(
+        "List of connector names to fire metrics for. Useful for when"
+        " wrapping dispatchers with many connectors, only a subset of which"
+        " should generate metrics. Defaults to all connectors.",
+        required=False, static=True)
```

Comment:
We probably don't need `op_mode` (because we publish based on conversation name or router name) or `metric_connectors` (because we're just using the messages to trigger publishing of counts).